### PR TITLE
Tret Industrial Complex + Away Site Exoplanet Generation

### PR DESCRIPTION
**Pull request by DreamySkrell**
image
image

image
image
image

### CODE REVIEW

**Comment by DreamySkrell** [comment]
!wip

**Comment by DreamySkrell** [comment]
!review

File: maps/_common/mapsystem/map.dm [hunk 1]
```diff
@@ -305,7 +305,14 @@
...
 	for (var/datum/map_template/template in selected)
-		if (template.load_new_z())
+		// get the theme if any
+		var/list/exoplanet_themes = null
+		var/datum/map_template/ruin/away_site/away_site = template
+		if(istype(away_site) && away_site.exoplanet_themes)
+			exoplanet_themes = away_site.exoplanet_themes
+
+		// actually load it
+		if (template.load_new_z(themes=exoplanet_themes))
```
Comment (FluffyGhoster) [inline maps/_common/mapsystem/map.dm:315]: As above

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

File: code/modules/maps/map_template.dm [hunk 2]
```diff
@@ -42,7 +42,10 @@
 	height = bounds[MAP_MAXY] - bounds[MAP_MINX] + 1
 	return bounds
 
-/datum/map_template/proc/load_new_z(var/no_changeturf = TRUE)
+/// `themes` should be assoc map of `/turf/unsimulated/marker/...` path to `/datum/exoplanet_theme/...` path.
+/// Exoplanet generation is not used if `themes` is null.
+/// See docs of `/datum/map_template/ruin/away_site`.
+/datum/map_template/proc/load_new_z(var/no_changeturf = TRUE, var/list/themes = null)
```
[the remark this answers is not in the document]
Comment (DreamySkrell) [reply code/modules/maps/map_template.dm]: fixed mmmaybe

File: maps/_common/mapsystem/map.dm (hunk 1 above)
Comment (DreamySkrell) [reply maps/_common/mapsystem/map.dm]: as above

File: code/modules/maps/map_template.dm [hunk 3]
```diff
@@ -77,6 +80,13 @@
 	smooth_zlevel(world.maxz)
 	resort_all_areas()
 
+	//exoplanet generation
+	if(themes)
+		for(var/marker_turf_type in themes)
+			var/datum/exoplanet_theme/exoplanet_theme_type = themes[marker_turf_type]
+			var/datum/exoplanet_theme/exoplanet_theme = new exoplanet_theme_type()
+			exoplanet_theme.generate_map(initial_z, 1, 1, 254, 254, marker_turf_type)
```
[the remark this answers is not in the document]
Comment (DreamySkrell) [reply code/modules/maps/map_template.dm]: as above

**Review by Arrow768 — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

**Pull request merged**